content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,351 @@
+
+## Handling requests within controller methods.
+
+Controllers are responsible for handling requests and sending responses. They are the middleman between the user and the model. They receive requests from the user and send responses back to the user. They also interact with the model to get the data it needs to perform its operations.
+
+Let's create a controller for our todo items. Run `rails generate controller todos`. This will create a controller file in the `app/controllers` folder. It will also create a folder named `todos` in the `app/views` folder. This folder will contain all the views for our todo items.
+
+To display a simple 'Hello World' message, we'll create a new method in the todos controller, and specify routes for it. Open the `app/controllers/todos_controller.rb` file and add the following code.
+
+```ruby
+class TodosController < ApplicationController
+  def index
+  end
+end
+```
+
+This is a simple controller method that will render the index view when a request is made to the `/todos` route. The index view is the default view for a controller. Rails will look for a view with the same name as the controller method. In this case, it will look for a view named `index.html.erb` in the `app/views/todos` folder.
+
+To create this view, create a new file named `index.html.erb` in the `app/views/todos` folder. Add the following code to the file.
+
+```html
+<h1>Hello World</h1>
+```
+
+Then add the following code to the `config/routes.rb` file.
+
+```ruby
+Rails.application.routes.draw do
+  get '/todos', to: 'todos#index'
+end
+```
+
+This will create a route that maps the `/todos` url to the `todos#index` controller method. The `get` method specifies the HTTP verb to use. The `to` option specifies the controller and method to use. The `todos#index` syntax is a shorthand for `controller: 'todos', action: 'index'`.
+
+Now run `rails s` or `bin/dev` to start the server, and visit `http://localhost:3000/todos` in your browser. You should see the 'Hello World' message.
+
+## Using ERB to write logic and display content in HTML.
+
+ERB stands for Embedded Ruby. It is a templating system that allows us to write ruby code within HTML files. This is useful because it allows us to write logic and display content in HTML. We can also use it to write HTML in a more concise way. 
+This aspect of Rails is the V in MVC. The view layer is responsible for displaying the content to the user. It is the part of the application that handles the presentation logic. The view layer interacts with the controller to get the data it needs to display. Without controllers, views would not be able to display any data.
+
+On our webpage, we would have a simple form for creating new todo items, and a table to list all the todo items. We'll use ERB to write the logic for this.
+
+Open the `app/views/todos/index.html.erb` file and add the following code.
```

Comment:
Should it be, "replace the content with the following"? Otherwise we still have a "Hello World" header.

Reply:
Yes, I'll fix that.